Gordon360/Controllers/LostAndFoundController.cs

```diff
@@ -0,0 +1,157 @@
+﻿using Gordon360.Authorization;
+using Gordon360.Models.CCT;
+using Gordon360.Models.CCT.Context;
+using Gordon360.Models.ViewModels;
+using Gordon360.Services;
+using Gordon360.Static.Names;
+using Microsoft.AspNetCore.Mvc;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+
+namespace Gordon360.Controllers
+{
+    [Route("api/[controller]")]
+    public class LostAndFoundController(CCTContext context, ILostAndFoundService lostAndFoundService) : GordonControllerBase
+    {
+
+        /// <summary>
+        /// Create a new missing item report with given data
+        /// </summary>
+        /// <param name="id">The id</param>
+        /// <returns>ObjectResult(ID) - An HTTP result code, with the ID of the created report if created successfully</returns>
+        [HttpPost]
+        [Route("missingitem")]
+        public ActionResult<int> CreateMissingItemReport([FromBody] MissingItemReportViewModel MissingItemDetails)
+        {
+            var authenticatedUserUsername = AuthUtils.GetUsername(User);
+
+            int ID = lostAndFoundService.CreateMissingItemReport(MissingItemDetails, authenticatedUserUsername);
+
+            return Ok(ID);
+        }
+
+        /// <summary>
+        /// Update Missing Item Report with the given id with given data
+        /// </summary>
+        /// <param name="id">The id of the report to update</param>
+        /// <returns>ObjectResult - the http status code result of the action, with the ID of the action taken</returns>
+        [HttpPost]
+        [Route("missingitem/{id}/actiontaken")]
+        public ActionResult<int> CreateActionTaken(int id, [FromBody] ActionsTakenViewModel ActionsTaken)
+        {
+            int ID = lostAndFoundService.CreateActionTaken(id, ActionsTaken);
+
+            return Ok(ID);
```

Comment:
It's confusing to have two different `id` variables that refer to ids of different objects. The route id should probably be `missingItemId`, and the returned Id should probably be `actionId`.

Reply:
Resolved in API #1111